src/components/sign-attributes/sign-attribute-pane.vue

```diff
@@ -0,0 +1,259 @@
+                        dropright
+                        @show="onValuesMenuShow()"
+                        @hide="onValuesMenuHide()"
+                    >
+                        <template v-slot:button-content>
+                            <span class="attr-name">{{attr.attributeName}}</span>
+                        </template>
+                        <b-dropdown-item
+                            v-for="attrValue in attr.values"
+                            :key="attrValue.id"
+                            @click="onAddAttribute(attr, attrValue)"
+                        >{{attrValue.value}}</b-dropdown-item>
+                    </b-dropdown>
+                </b-dropdown>
+            </li>
+            <li
+                class="pr-2"
+                v-for="attribute in attributes"
+                :key="attribute.interpretationAttributeId"
+            >
+                <sign-attribute
+                    :attribute="attribute"
+                    @attribute-click="onAttributeClick(attribute)"
+                />
+            </li>
+        </ul>
+        <sign-attribute-modal />
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Emit } from 'vue-property-decorator';
+import { Artefact } from '@/models/artefact';
+import {
+    TextFragmentData,
+    TextFragment,
+    SignInterpretation,
+    ArtefactTextFragmentData,
+} from '@/models/text';
+import TextFragmentComponent from '@/components/text/text-fragment.vue';
+import { EditionInfo } from '@/models/edition';
+import { AttributeDTO, AttributeValueDTO, InterpretationAttributeDTO } from '@/dtos/sqe-dtos';
+import SignAttribute from './sign-attribute.vue';
+import SignAttributeModal from './sign-attribute-modal.vue';
+import { TextFragmentAttributeOperation } from '@/views/artefact-editor/operations';
+import { BDropdown, BvEvent } from 'bootstrap-vue';
+
+@Component({
+    name: 'sign-attribute-pane',
+    components: {
+        'sign-attribute-modal': SignAttributeModal,
+        'sign-attribute': SignAttribute,
+    },
+})
+export default class SignAttributePane extends Vue {
+    private keepOpen = false;
+    private attributesMenu: AttributeDTO[] = [];
+
+    public get artefactEditor() {
+        return this.$state.artefactEditor;
+    }
+    public get selectedSignsInterpretation(): SignInterpretation[] {
+        return this.artefactEditor.selectedSignsInterpretation;
+    }
+
+    private get attributesMetadata() {
+        return (
+            this.$state.editions.current?.attributeMetadata?.allAttributes || []
+        );
+    }
+
+    public get attributes(): InterpretationAttributeDTO[] {
+        let attributeValues: number[] = [];
+        let first = true;
+
+        if (!this.selectedSignsInterpretation.length) {
+            return [];
+        }
+
+        for (const si of this.selectedSignsInterpretation) {
+            const siValues = [
+                ...si.attributes.map((attr) => attr.attributeValueId),
+            ];
```

Comment:
This whole function took me a little while to read through.  Everyone reads things differently, so I don't know that you need to change this but, for instance, I would prefer simply `const siValues = si.attributes.map((attr) => attr.attributeValueId)` rather than running the map function and then destructuring the returned array into another array. I get the sense that this whole for loop could be done maybe with a single map, or maybe a few successive ones, but I leave it to you to decide what is most readable.

Reply:
I've added some comments and removed the unnecessary [...] . Basically we're just intersection all the attributes from all the sign interpretations.